src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2518,11 +2565,12 @@ private static abstract class LocalMutationRunnable implements Runnable
         public final void run()
         {
             final Verb verb = verb();
-            long nowNanos = MonotonicClock.Global.approxTime.now();
+            approxStartTimeNanos = MonotonicClock.Global.approxTime.now();
```

Comment:
Note: Do we need to make this `volatile` to ensure visibility for the start time for the thread querying the virtual table?